functions-and-scope.js

```diff
@@ -0,0 +1,127 @@
+// Je gaat functies schrijven die we kunnen hergebruiken om een lijst met eindcijfers van studenten te checken. Je zult over de cijfers heen moeten itereren (hoe pak je dat aan?),
+// maar ook een manier moeten vinden om hetgeen dat je verzamelt ergens te bundelen. Op deze manier zul je ontdekken hoe je omgaat met scope. Pak vooral het hoofdstuk op EdHub over for-loops er nog eens bij!
+// Tip: je mag hier geen ingebouwde object methoden gebruiken, dus daar hoef je niet naar te kijken.
+
+const grades = [9, 8, 5, 7, 7, 4, 9, 8, 8, 3, 6, 8, 5, 6];
+
+/* Opdracht  1: Cum Laude */
+
+/* 1a: Script schrijven  */
+// De administratie moet weten hoeveel studenten er dit blok cum laude zijn afgestudeerd (8 of hoger). Daar moeten namelijk speciale diploma's voor besteld worden.
+// Schrijf de stapjes om dit te kunnen checken eerst uit en vraag jezelf de volgende dingen af:
+// * Hoe kan ik iedere waarde van de array checken op deze conditie?    for loop
+// * Hoe zorg ik ervoor dat dit ook werkt wanneer de array 100 entries bevat?   array als argument gebruiken
+// * Hoe zorgt ik ervoor dat wanneer ik een cijfer tegenkom die aan de conditie voldoet, ik dit ergens kan bijhouden?   push() naar nieuwe array
+// Log het antwoord in de terminal.
+
+// ---- Verwachte uitkomst: 6
+
+
+function CumLaude(array) {
+    let passed = [];
+    for(let i=0;i<array.length;i++) {
+        if(array[i] >= 8) {
+            passed.push(array[i])
+        }
+    }
+    return passed.length;
+}
+console.log(CumLaude(grades))
```

Comment:
Werkt keurig. Je zou ook een variabele kunnen maken die een nummer verwacht. Vervolgens wanneer de if-statement in de for-loop true is, tel je 1 bij het getal op. Zo kom je ook op het totaal aantal studenten die een Cum Laude hebben.